### PR TITLE
Books api

### PR DESCRIPTION
**Pull request by eledwinn**
## Summary
1. Getting books and books details from API

## Screenshot
NA

## Trello Card
https://trello.com/c/1x4BADBA/29-books-api

### CODE REVIEW

File: src/core/auth/session.service.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
 'use strict';
 
-angular.module('core.auth').factory('Session', function () {
+angular.module('core.auth').factory('Session', ['$http', function ($http) {
```
Comment (FrankIglesias) [inline src/core/auth/session.service.js]: use arrow

File: src/core/auth/session.service.js [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
   const localTokenInfo = localStorage.getItem('tokenInfo');
   const localUser = localStorage.getItem('user');
   if (localTokenInfo) {
     this.tokenInfo = JSON.parse(localTokenInfo);
+    updateSession();
   }
   if (localUser) {
     this.user = JSON.parse(localUser);
   }
+
```
Comment (FrankIglesias) [inline src/core/auth/session.service.js]: delete this lne

File: src/core/book/book.service.js [hunk 3]
```diff
@@ -1,9 +1,34 @@
 'use strict';
 
-angular.module('core.book').factory('Books', ['$http', function ($http) {
+angular.module('core.book').factory('Books', ['$http', 'Environment', function ($http, Environment) {
+  const baseUrlApi = Environment.wBooksApiUrl + 'books';
+  const mapper = (book) => {
```
Comment (FrankIglesias) [inline src/core/book/book.service.js]: remove parenthesis

File: src/core/book/book.service.js [hunk 4]
```diff
@@ -1,9 +1,34 @@
 'use strict';
 
-angular.module('core.book').factory('Books', ['$http', function ($http) {
+angular.module('core.book').factory('Books', ['$http', 'Environment', function ($http, Environment) {
```
Comment (FrankIglesias) [inline src/core/book/book.service.js]: arrow

File: src/core/book/book.service.js [hunk 5]
```diff
@@ -1,9 +1,34 @@
...
+      return $http.get(baseUrlApi).then(resp => {
+        if (resp.data) {
+          return resp.data.map(mapper);
+        }
+        return resp.data;
+      });
+    },
+    information: (id) => {
+      return $http.get(`${baseUrlApi}/${id}`).then(resp => {
+        if (resp.data) {
```
Comment (FrankIglesias) [inline src/core/book/book.service.js]: use a ternary

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: src/core/auth/session.service.js (hunk 1 above)
Comment (eledwinn) [reply src/core/auth/session.service.js]: Is a angular service, is creating a new context.

File: src/core/book/book.service.js [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+      description: book.description
+    };
+  };
+
   return {
     get: () => {
-      return $http.get('assets/books.json', { cache: true }).then(resp => {
-        return resp.data;
+      return $http.get(baseUrlApi).then(resp => {
+        return resp.data ? resp.data.map(mapper) : resp.data;
```
Comment (FrankIglesias) [inline src/core/book/book.service.js:21]: this can be done with `&&` operator

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**